nix/modules/environment.nix

```diff
@@ -0,0 +1,59 @@
+{ lib, config, pkgs, ... }:
+
+{
+  options.environment = {
+    systemPackages = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+    };
+  };
+
+  config =
+    let
+      pathDir = "/run/system-manager/sw";
+    in
+    {
+      environment.etc = {
+        "profile.d/system-manager-path.sh".source =
+          pkgs.writeText "system-manager-path.sh" ''
+            export PATH=${pathDir}/bin/:''${PATH}
+          '';
+
+        # TODO: figure out how to properly add fish support. We could start by
+        # looking at what NixOS and HM do to set up the fish env.
+        #"fish/conf.d/system-manager-path.fish".source =
+        #  pkgs.writeTextFile {
+        #    name = "system-manager-path.fish";
+        #    executable = true;
+        #    text = ''
+        #      set -gx PATH "${pathDir}/bin/" $PATH
+        #    '';
+        #  };
+      };
+
+      systemd.services.system-manager-path = {
+        enable = true;
+        description = "";
+        wantedBy = [ "system-manager.target" ];
+        serviceConfig = {
+          Type = "oneshot";
+          RemainAfterExit = true;
+        };
+        script =
+          let
+            pathDrv = pkgs.buildEnv {
+              name = "system-manager-path";
+              paths = config.environment.systemPackages;
+              pathsToLink = [ "/bin" ];
+            };
+          in
+          ''
+            mkdir --parents $(dirname "${pathDir}")
```

Comment:
I think `/run/system-manager` should be a symlink to make it possible to put stuff under `/run/system-manager` while keeping atomic semantics when updating.

Reply:
Hmm, what else would you put there?
The situation is slightly different here compared to nixos since we don't have an activation script that gets called during bootup, so we need to rely on systemd to set up everything for us, so I'm not sure we would have any use for most of the stuff that nixos puts directly under `/run/current-system`, but I'd be happy to be proven wrong,

Reply:
> we don't have an activation script that gets called during bootup

Maybe we should have one, e.g. to be able to `nix copy` system-manager's closure and activate it over ssh.

Reply:
That's already supported, with an interface very similar to `nixos-rebuild`. The thing is that for system-manager we kind of merged the concepts of activation script and `switch-to-configuration`, since the distinction does not make sense in our case (in nixos, the difference is that when you boot the machine, you only run the activation script, but when you switch a running system you run `switch-to-configuration` which will in turn run the activation script).

My point was rather that the toplevel `current-system` dir contains things like the kernel, initrd, activate script, specialisations, etc, none of which are directly useful in the context of system-manager. At least not currently.
So if you say that the system-manager dir should be a symlink, then I wonder what it should point at, since just having another store path with only the symlink to `sw` inside does not seem very useful to me.
If we make it a simple directory now, we can still make it a symlink later if we discover a use for it.

Reply:
Everyone satisfied to resolve this thread? It would be great to have this PR merged.